lib/std/fs/test.zig

```diff
@@ -1647,6 +1647,39 @@ test "open file with exclusive nonblocking lock twice (absolute paths)" {
     try testing.expectError(error.WouldBlock, file2);
 }
 
+test "read from locked file" {
+    var random_bytes: [12]u8 = undefined;
+    std.crypto.random.bytes(&random_bytes);
+
+    var random_b64: [fs.base64_encoder.calcSize(random_bytes.len)]u8 = undefined;
+    _ = fs.base64_encoder.encode(&random_b64, &random_bytes);
+
+    const basename = random_b64 ++ "-zig-test-read-locked-file.txt";
+
+    {
+        const f = try std.fs.cwd().createFile(basename, .{ .read = true });
+        defer f.close();
+        var buffer: [1]u8 = undefined;
+        _ = try f.readAll(&buffer);
+    }
+    {
+        const f = try std.fs.cwd().createFile(basename, .{
+            .read = true,
+            .lock = .exclusive,
+        });
+        defer f.close();
+        const f2 = try std.fs.cwd().openFile(basename, .{});
+        defer f2.close();
+        var buffer: [1]u8 = undefined;
+        if (builtin.os.tag == .windows) {
+            try std.testing.expectError(error.LockViolation, f2.readAll(&buffer));
+        } else {
+            try std.testing.expectEqual(0, f2.readAll(&buffer));
+        }
+    }
+    try std.fs.cwd().deleteFile(basename);
```

Comment:
Why is this test creating files in the cwd instead of using `std.testing.tmpDir`? Even better would be to use `testWithAllSupportedPathTypes` (see other tests that use it for an example).